ci-operator/config/openshift-eng/rebasebot/openshift-eng-rebasebot-main.yaml

```diff
@@ -6,7 +6,7 @@ build_root:
       # Install go toolset
       USER root
 
-      ENV GO_VERSION=1.20
+      ENV GO_VERSION=1.21.1
```

Comment:
Could we set this env var on the job for `cloud-provider-azure` instead of everywhere?
The other projects are still all being built by 1.20, so it's probably best not to touch their go.mod/go.sum/vendor with go 1.21.